app/controllers/mixins/middleware_common_mixin.rb

```diff
@@ -25,4 +29,35 @@ def clear_topology_breadcrumb
       @breadcrumbs.clear
     end
   end
+
+  def init_show(model_class = self.class.model)
+    clear_topology_breadcrumb
+    @lastaction = 'show'
+    @showtype = 'main'
+    @gtl_url = '/show'
+    @display = params[:display] || 'main' unless control_selected?
+    @record = identify_record(params[:id], model_class)
+  end
+
+  def show_middleware
+    return if record_no_longer_exists?(@record)
+    drop_breadcrumb({:name => display_name,
+                     :url  => "/#{controller_name}/show_list?page=#{@current_page}&refresh=y"},
+                    true)
+    breadcrumb_url = "/#{controller_name}/show/#{@record.id}?refresh=n&display=#{@display}"
+    case @display
+    when 'download_pdf', 'main', 'summary_only' then show_summary(@display, controller_name, @record)
+    when 'timeline' then show_timeline(@record, breadcrumb_url)
+    when 'performance' then show_performance(breadcrumb_url)
+    end
+  end
+
+  def show_middleware_display(record, clazz, alt_controller_name = nil)
+    display = clazz.name.underscore
```

Comment:
Random thing, but the proper way to spell class when you need it as a variable in ruby is `klass`, not `clazz`...

(We're using `klass` 1495 times, `clazz` 4 times)